### ISSUE TITLE
@aws-sdk/client-ssm 3.418.0 returns binary data

### ISSUE THREAD

**Issue by vladaman** [aws/aws-sdk-js-v3#5251]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

When calling SSM We get the following exception:
```
9589 |     };
9590 | }
9591 | const parseBody = (streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
9592 |     // console.log('xencoded',Buffer.from(encoded).toString('base64'));
// Logged Base64 Value here is: 34vvv70IAAAAAAAAAO+/ve+/vU1PAjEQ77+977+9y5zvv73vv73vv71ucXsz77+9AwkS77+9xovvv70w0qIN77+977+977+9KSAS77+977+977+9Bg8c1JB477+977+977+9eSbvv70eJu+/vQZb77+9bzFgZ++/vQkE77+9ae+/ve+/vcm877+9y7sZKO+/ve+/vSvckiLvv71Z77+977+977+977+9UDLvv71iXO+/vXFT77+977+977+977+9d++/vQ9OZxojPiMZyrod77+977+92avvv70ICVxx77+977+977+90bwP77+9FCnvv704be+/ve+/vWhu77+9KVLvv73vv71VUVdS77+977+9de+/vQzvv71877+9K++/vQjvv73vv73vv73vv70vPD9i77+9Hu+/ve+/ve+/vXTvv70pXSHvv73Sru+/ve+/ve+/vVDvv73vv711PT87JO+/vSfvv71dOFdSXu+/vWUu77+977+9PO+/vX/vv70UIu+/ve+/vVbvv73vv70nDQEh77+977+9AQAA
9593 |     if (encoded.length) {
9594 |         return JSON.parse(encoded);
                   ^
SyntaxError: JSON Parse error: Unrecognized token 'ߋ'
  Deserialization error: to see the raw response, inspect the hidden field {error}.$response on this object.
      at /home/vlad/workspace/lambda/fk-bun-test/node_modules/@aws-sdk/client-ssm/dist-cjs/protocols/Aws_json1_1.js:9594:15
      at processTicksAndRejections (:1:2602)
```

### SDK version number

@aws-sdk/client-ssm": "^3.418.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

bun -v 1.0.1
aws-sdk-js/3.418.0 ua/2.0 os/linux#6.2.0-33-generic lang/js md/nodejs#18.15.0 api/ssm#3.418.0

### Reproduction Steps

```typescript
 public async getParameters(names: string[]): Promise<string[]> {
        const params = {
            Names: names,
            // WithDecryption: true
        };
        try {
            const command = new GetParametersCommand(params);
            const response = await this.client.send(command);
            console.log('parameter', response);
            return response.Parameters?.map((p) => {
                return p.Value;
            });
        } catch (e) {
            console.log(e);
            return [];
        }
    }
```

Base64 encoded 

### Observed Behavior

post payload:
```javascript
{
  protocol: "https:",
  hostname: "ssm.eu-west-1.amazonaws.com",
  port: undefined,
  method: "POST",
  path: "/",
  headers: {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": "AmazonSSM.GetParameters"
  },
  body: "{\"Names\":[\"/prod/databases/mysql/host\",\"/prod/databases/mysql/port\"]}"
}
```

Server responded properly:
```
    [Symbol(#req)]: Response (237 bytes) {
      ok: true,
      url: "https://ssm.eu-west-1.amazonaws.com/",
      headers: Headers {
        "date": "Sat, 23 Sep 2023 09:33:42 GMT",
        "content-type": "application/x-amz-json-1.1",
        "content-length": "237",
        "connection": "keep-alive",
        "content-encoding": "gzip",
        "server": "Server",
        "x-amzn-requestid": "90c38ae3-9d35-4a9b-9cbf-6d95b67a6b1f"
      },
      statusText: "OK",
      redirected: true,
      bodyUsed: false,
      status: 200,
      Blob (237 bytes)
    },
```

```

### Expected Behavior

Return proper object

### Possible Solution

_No response_

### Additional Information/Context

_No response_

**Comment by RanVaknin** [comment]
Hi @vladaman ,

Is this response coming from the SSM service? Because Im not familiar with this log format. Is that response coming back from a proxy? I'm not able to reproduce it myself:

```js
import { SSMClient, GetParametersCommand } from '@aws-sdk/client-ssm'

const client = new SSMClient({
    region: "us-east-1"
})

client.middlewareStack.add(next => async(args) => {
    const response = await next(args);
    console.log(response);
    return response
}, {step: 'build'})

async function getParameters(names) {
    const params = {
        Names: names,
    };
    try {
        const command = new GetParametersCommand(params);
        const response = await client.send(command);
        console.log('parameter', response);
        return response.Parameters?.map((p) => {
            return p.Value;
        });
    } catch (e) {
        console.log(e);
        return [];
    }
}
getParameters(["foo"])
```
Output:

```ruby
response: HttpResponse {
    statusCode: 200,
    reason: 'OK',
    headers: {
      server: 'Server',
      date: 'Mon, 25 Sep 2023 03:42:43 GMT',
      'content-type': 'application/x-amz-json-1.1',
      'content-length': '45',
      connection: 'keep-alive',
      'x-amzn-requestid': 'REDACTED'
    },
    body: IncomingMessage {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      socket: null,
      httpVersionMajor: 1,
      httpVersionMinor: 1,
      httpVersion: '1.1',
      complete: true,
      rawHeaders: [Array],
      rawTrailers: [],
      joinDuplicateHeaders: undefined,
      aborted: false,
      upgrade: false,
      url: '',
      method: null,
      statusCode: 200,
      statusMessage: 'OK',
      client: [TLSSocket],
      _consuming: true,
      _dumped: false,
      req: [ClientRequest],
      [Symbol(kCapture)]: false,
      [Symbol(kHeaders)]: [Object],
      [Symbol(kHeadersCount)]: 12,
      [Symbol(kTrailers)]: null,
      [Symbol(kTrailersCount)]: 0
    }
  },
  output: {
    '$metadata': {
      httpStatusCode: 200,
      requestId: 'REDACTED',
      extendedRequestId: undefined,
      cfId: undefined,
      attempts: 1,
      totalRetryDelay: 0
    },
    InvalidParameters: [ 'foo' ],
    Parameters: []
  }
```

My suspicion is that the response you are getting back is gzip encoded and the one Im getting is not (or at least the header isn't sent)

Can you give us a little more information about how you are getting these request and response logs?

Thanks!
Ran~

**Comment by vladaman** [comment]
OK, I confirm, that this has been fixed by upgrading to bun 1.0.3
